src/java/org/apache/cassandra/cql3/Constants.java

```diff
@@ -44,7 +45,19 @@
 
     public enum Type
     {
-        STRING,
+        STRING
+        {
+            public AbstractType<?> getPreferedTypeFor(String text)
+            {
+                 if(Charset.forName("US-ASCII").newEncoder().canEncode(text))
+                 {
+                     return AsciiType.instance;
+                 } else
```

Comment:
nit: The `else` should be on the next line. You can also choose to remove the `else` as it is not needed do to the `return`.